docs(cn): translate reference/react-dom/createPortal into Chinese (#1140)
reactjs/zh-hans.react.dev: src/content/reference/react-dom/createPortal.md

</div>
```

[请查看以下更多示例](#usage)。

portal 只改变 DOM 节点的所处位置。在其他方面，传入 portal 中的 JSX 将作为渲染它的 React 组件的子节点。该子节点可以访问由父节点树提供的 context 对象、事件将从子节点冒泡到父节点树，以及遵循 React 树的规则。

#### 参数 {/*parameters*/}

* `children`：可以使用 React 渲染的任何内容，例如 JSX 片段（例如 `<div />` 或 `<SomeComponent />`）、[Fragment](/reference/react/Fragment)（`<>...</>`）、字符串或数字，或者是这些内容的数组。

* `domNode`：某个 DOM 节点，例如由 `document.getElementById()` 返回的节点。节点必须已经存在。在更新过程中传递不同的 DOM 节点将导致 portal 内容被重新创建。

#### 返回值 {/*returns*/}

`createPortal` 返回一个 React 节点，该节点可以包含在 JSX 中或从 React 组件中返回。如果 React 在渲染输出中遇见它，它将把提供的 `children` 放入提供的 `domNode` 中。

#### 警告 {/*caveats*/}

* portal 中的事件传播遵循 React 树而不是 DOM 树。例如，如果你在 portal 中触发点击事件，并且 portal 被包装在 `<div onClick>` 中，则将触发 `onClick` 处理程序。如果这会导致问题，请在 portal 内部停止事件传播，或将 portal 本身移动到 React 树中的上层。

---

## 用法 {/*usage*/}

### 渲染到 DOM 的不同部分 {/*rendering-to-a-different-part-of-the-dom*/}

> *Portals* 允许组件将它们的某些子元素渲染到 DOM 中的不同位置。这使得组件的一部分可以“逃脱”它所在的容器。例如，组件可以显示出现在页面其余部分之上和之外的模态对话框或工具提示。

Reviewer: ```suggestion
*portal* 允许组件将它们的某些子元素渲染到 DOM 中的不同位置。这使得组件的一部分可以“逃脱”它所在的容器。例如，组件可以显示出现在页面其余部分之上和之外的模态对话框或工具提示。
```

> 要创建 portal，请使用 <CodeStep step={1}>JSX</CodeStep> 和 <CodeStep step={2}>应该放置的 DOM 节点</CodeStep> 渲染 `createPortal` 的结果：

Reviewer 2: 不知道是不是因为夹杂了 <CodeStep> 的原因

</body>
```

> portal 只改变 DOM 节点的所处位置。在其他方面，入 portal 中的 JSX 将作为渲染它的 React 组件的子节点。该子节点可以访问由父节点树提供的 context 对象、事件将仍然从子节点冒泡到父节点树。

Reviewer 2: ```suggestion
portal 只改变 DOM 节点的所处位置。另一方面，portal 中的 JSX 将作为实际渲染它的 React 组件的子节点。该子节点可以访问由父节点树提供的 context 对象、事件将仍然从子节点冒泡到父节点树。
```

---

### 使用 portal 渲染模态对话框 {/*rendering-a-modal-dialog-with-a-portal*/}

> 你可以使用 portal 创建一个浮动在页面其余部分之上的模态对话框，即使呼出对话框的组件位于带有 `overflow: hidden` 或其他干扰对话框的样式的容器中。

Reviewer 2: ```suggestion
你可以使用 portal 创建一个浮动在页面其余部分之上的模态对话框，即使呼出对话框的组件位于带有 `overflow: hidden` 或其他干扰对话框样式的容器中。
```

<Pitfall>

使用 portal 时，确保应用程序的无障碍性非常重要。例如，你可能需要管理键盘焦点，以便用户可以以自然的方式进出 portal。

> 创建模态对话框时，请遵循 [WAI-ARIA 模态作者实践指南](https://www.w3.org/WAI/ARIA/apg/#dialog_modal)。如果你使用了社区包，请确保它是无障碍的，并遵循这些指南。

Reviewer 2: ```suggestion
创建模态对话框时，请遵循 [WAI-ARIA 模态实践指南](https://www.w3.org/WAI/ARIA/apg/#dialog_modal)。如果你使用了社区包，请确保它是无障碍的，并遵循这些指南。
```

</Pitfall>

---

### 将 React 组件渲染到非 React 服务器标记中 {/*rendering-react-components-into-non-react-server-markup*/}

> 如果你的网站只有一部分使用 React 构建，而其他部分是静态页面或由服务器呈现的页面，则 portal 可能非常有用。如果你的页面使用 Rails 等服务端框架构建，则可以在静态区域（例如侧边栏）中创建交互区域。与拥有 [多个独立的 React 根](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) 相比，portal 将应用程序视为具有共享状态的单个 React 树，即使其部分呈现到 DOM 的不同部分也是如此。

Reviewer 2: ```suggestion
如果你在静态或服务端渲染的网站中只有某一部分使用 React，则 portal 可能非常有用。如果你的页面使用 Rails 等服务端框架构建，则可以在静态区域（例如侧边栏）中创建交互区域。与拥有 [多个独立的 React 根](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) 相比，portal 将应用程序视为具有共享状态的单个 React 树，即使其部分呈现到 DOM 的不同部分也是如此。
```
